pkg/diff/sql_generator.go

```diff
@@ -1228,6 +1288,176 @@ func (a *attachPartitionSQLVertexGenerator) GetDeleteDependencies(_ schema.Table
 	return nil
 }
 
+type sequenceSQLVertexGenerator struct {
+	deletedTablesByName map[string]schema.Table
+	tableDiffsByName    map[string]tableDiff
+}
+
+func (s *sequenceSQLVertexGenerator) Add(seq schema.Sequence) ([]Statement, error) {
+	sb := strings.Builder{}
+	sb.WriteString(fmt.Sprintf("CREATE SEQUENCE %s\n", seq.GetFQEscapedName()))
+	sb.WriteString(fmt.Sprintf("\tAS %s\n", seq.Type))
+	sb.WriteString(fmt.Sprintf("\tINCREMENT BY %d\n", seq.Increment))
+	sb.WriteString(fmt.Sprintf("\tMINVALUE %d MAXVALUE %d\n", seq.MinValue, seq.MaxValue))
+	cycleModifier := ""
+	if !seq.Cycle {
+		cycleModifier = "NO "
+	}
+	sb.WriteString(fmt.Sprintf("\tSTART WITH %d CACHE %d %sCYCLE\n", seq.StartValue, seq.CacheSize, cycleModifier))
+
+	var hazards []MigrationHazard
+	if seq.Owner == nil {
+		hazards = append(hazards, migrationHazardSequenceCannotTrackDependencies)
+	}
+
+	return []Statement{{
+		DDL:     sb.String(),
+		Timeout: statementTimeoutDefault,
+		Hazards: hazards,
+	}}, nil
+}
+
+func (s *sequenceSQLVertexGenerator) Delete(seq schema.Sequence) ([]Statement, error) {
+	var hazards []MigrationHazard
+	if seq.Owner != nil && (s.isDeletedWithOwningTable(seq) || s.isDeletedWithColumns(seq)) {
+		return nil, nil
+	} else if seq.Owner == nil {
+		hazards = append(hazards, migrationHazardSequenceCannotTrackDependencies)
+	}
+	return []Statement{{
+		DDL:     fmt.Sprintf("DROP SEQUENCE %s", seq.GetFQEscapedName()),
+		Timeout: statementTimeoutDefault,
+		Hazards: hazards,
```

Comment:
Dropping a sequence will destroy state (the internal value), so there should be a hazard for that right? This should trigger on recreates in the absence of alters.